### ISSUE TITLE
Improvements to current plugin

### ISSUE THREAD

**Issue by rtmaster** [GridProtectionAlliance/osisoftpi-grafana#6]
I would like to make a couple of improvements and I am looking for your advice on how to achieve them:

- Whenever you specify a Default AF Database in the configuration of the data source, Grafana should allow you to select only that particular AF when browsing a metric in a panel. This will allow to manage database-level segregation (you don't want to expose your full AF Server to a specific set of Grafana users

- Whenever you set a new metric with PI datasource, the hierarchy should appear already populated up to the AF Database (no need to manually write the name of AF server and Toggle Edit Mode)

- Resolve the bug of PI AF Element disappearing when editing existing panel

Please @ritchiecarroll and @EEParker  can you tell me what you think about the above and are you open for a discussion?

**Comment by alqassak** [comment]
Hi @rtmaster 

We were able to resolve the bugs that you are mentioning, please apply the below changes in the plugin files and it will work for you,

In file osisoftpi-grafana-master\dist\query_ctrl.js  And osisoftpi-grafana-master\dist\query_ctrl.js.map 
**This statement**
`_this.target.target = _this.target.target || 'select element';`
 **Was replaced with**
`_this.target.target = _this.target.target || ‘;';`

In file osisoftpi-grafana-master\dist\query_ctrl.js  
**This statement** 
`ctrl.segments = ctrl.segments.splice(0, fromIndex);
                  if (ctrl.segments[ctrl.segments.length - 1].expandable) {
                    ctrl.segments.push(ctrl.uiSegmentSrv.getSegmentForValue(null, "Select AF Database"));
                  }`
**Was replaced with**
`ctrl.segments = ctrl.segments.splice(0, fromIndex+1);
                  if (ctrl.segments[ctrl.segments.length - 1].expandable) {
                    ctrl.segments.push(ctrl.uiSegmentSrv.getSegmentForValue(null, "Select AF Database"));
}`

And osisoftpi-grafana-master\dist\query_ctrl.js.map
**This statement**
`ctrl.segments = ctrl.segments.splice(0, fromIndex)\n          if (ctrl.segments[ctrl.segments.length - 1].expandable) {\n            ctrl.segments.push(ctrl.uiSegmentSrv.getSegmentForValue(null, \"Select AF Database\"))\n          }\n`
**Was replaced with**
`ctrl.segments = ctrl.segments.splice(0, fromIndex+1)\n          if (ctrl.segments[ctrl.segments.length - 1].expandable) {\n            ctrl.segments.push(ctrl.uiSegmentSrv.getSegmentForValue(null, \"Select AF Database\"))\n          }\n`

**Comment by ritchiecarroll** [comment]
@alqassak Do you want to provide a pull-request for this fix?

**Comment by alqassak** [comment]
@ritchiecarroll sure,

**Comment by ritchiecarroll** [comment]
Fixed with #9

**Comment by rtmaster** [comment]
Hey @ritchiecarroll thanks for merging @alqassak PR.
I tested your fix today and items 2 and 3 seem to work fine with my Grafana 4.4.1 version.

I noticed that item 1 was not part of the package (Whenever you specify a Default AF Database in the configuration of the data source, Grafana should allow you to select only that particular AF when browsing a metric in a panel)

Can I open a separate issue for that? Or did I miss something?
Thanks again for the great contributions
